### PR TITLE
feat: add support for .trivyignore.yaml

### PR DESCRIPTION
**Pull request by knqyf263**
## Description
Support the structured ignore file.

### Example:

```
$ cat .trivyignore.yaml
vulnerabilities:
  - id: CVE-2023-30861
    paths:
      - "pkg/fanal/analyzer/language/python/poetry/testdata/happy/poetry.lock"
      - "pkg/fanal/analyzer/language/python/pip/testdata/requirements.txt"
    statement: Accept the risk
  - id: CVE-2023-37920
  - id: GHSA-5crp-9r3c-p9vr
     expired_at: 2023-09-01

misconfigurations:
  - id: AVD-DS-0002
  - id: AVD-DS-0003
     paths:
       - docs/Dockerfile
     statement: The container image is not deployed
```

It is still experimental since we're aware of more requirements around ignoring and possibly adding breaking changes. Thus, the file path must be explicitly specified as below so that it will not break compatibility.

```
$ trivy repo --ignorefile .trivyignore.yaml .
```

If the file extension is `.yml` or `.yaml`, Trivy tries to parse that in the YAML format. Otherwise, it is treated as the legacy `.trivyignore`.

## TODOs
- [x] Support .trivyignore.yaml
- [x] Backward compatibility
- [ ] Write tests
- [ ] Write docs

## Related issues
- Close https://github.com/aquasecurity/trivy/issues/4022

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [ ] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: pkg/result/ignore.go [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+	Paths []string `yaml:"paths"`
+
+	// ExpiredAt is the expiration date of the ignore finding.
+	// If ExpiredAt is not set, the ignore finding is always valid.
+	// required: false
+	ExpiredAt time.Time `yaml:"expired_at"`
+
+	// Statement describes the reason for ignoring the finding.
+	// required: false
+	Statement string `yaml:"statement"`
```
Comment (DmitriyLewen) [inline pkg/result/ignore.go:39]: yaml files support comments. 
Do you think we need this field?

e.g.:
```yaml
vulnerabilities:
  # Accept the risk
  - id: CVE-2023-30861
    paths:
      - "pkg/fanal/analyzer/language/python/poetry/testdata/happy/poetry.lock"
      - "pkg/fanal/analyzer/language/python/pip/testdata/requirements.txt"
```

File: pkg/result/ignore.go [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+}
+
+type IgnoreFindings []IgnoreFinding
+
+func (findings IgnoreFindings) Match(path, id string) bool {
+	for _, finding := range findings {
+		if len(finding.Paths) != 0 && !slices.Contains(finding.Paths, path) {
```
Comment (DmitriyLewen) [inline pkg/result/ignore.go:48]: Perhaps we want to use `wildcard` for paths?

Comment (DmitriyLewen) [reply pkg/result/ignore.go:48]: don't forget: we need to update docs if we don't add wildcards.

File: pkg/result/ignore.go (hunk 1 above)
Comment (knqyf263) [reply pkg/result/ignore.go:41]: Yes, we want to generate VEX from .trivyignore.yaml in the future. VEX has `impact_statement`, so we should have a corresponding field.
https://github.com/openvex/spec/blob/main/OPENVEX-SPEC.md#statement-fields

File: pkg/result/ignore.go (hunk 2 above)
Comment (knqyf263) [reply pkg/result/ignore.go]: Added https://github.com/aquasecurity/trivy/pull/5070/commits/c624be43c9c7339e37ac7c2a7974599d4cbbfe78

**Review by DmitriyLewen — APPROVED** [review]
lgtm

**Pull request merged**